src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+                        currentBaseQuality.getMappingErrorProbability() / 4);
+                }
+            }
+        }
+
+        switch (hetPriorType) {
+
+            case BALANCED:
+                /* set the allele fraction to 1/2 */
+                refAltLogLikelihood = getRefAltHetLogLikelihoodFixedAlleleFraction(0.5, alphaList, betaList);
+                break;
+
+            case HETEROGENEOUS:
+                /* marginalize allele fraction by integrating {@link getRefAltHetLogLikelihoodFixedAlleleFraction} over
+                   the precomputed allele fraction prior */
+                ArrayList<Double> refAltLogLikelihoodList = new ArrayList<>(gaussIntegrationAbscissas.size());
+                refAltLogLikelihoodList.addAll(
+                        gaussIntegrationAbscissas.stream()
+                                .mapToDouble(Double::doubleValue)
+                                .map(f -> getRefAltHetLogLikelihoodFixedAlleleFraction(f, alphaList, betaList))
+                                .boxed().collect(Collectors.toList()));
+                ArrayList<Double> logLikelihoodIntegrandWithPriorAndWeights = new ArrayList<>(gaussIntegrationAbscissas.size());
+                logLikelihoodIntegrandWithPriorAndWeights.addAll(
+                        IntStream.range(0, gaussIntegrationAbscissas.size())
+                                .mapToDouble(i -> refAltLogLikelihoodList.get(i) + gaussIntegrationLogWeights.get(i) +
+                                        alleleFractionLogPriors.get(i))
+                                .boxed().collect(Collectors.toList()));
+                refAltLogLikelihood = GATKProtectedMathUtils.naturalLogSumExp(logLikelihoodIntegrandWithPriorAndWeights);
+                break;
+
+            default: /* we shouldn't be here anyway */
+                refAltLogLikelihood = 0.0;
+
+        }
+
+        return errorLogLikelihood + refAltLogLikelihood;
+    }
+
+    /**
+     * wrapper methods
+     */
+
+    /**
+     * Returns map of base-pair to error probabilities at a given locus. All reads are considered (not just ACTG)
+     * @param locus locus
+     * @return map of base-pair to error probabilities
+     */
+    private Map<Nucleotide, ArrayList<BaseQuality>> getPileupBaseQualities(final SamLocusIterator.LocusInfo locus) {
+
+        Map<Nucleotide, ArrayList<BaseQuality>> baseQualities = new HashMap<>();
+
+        /* make sure that we have keys at least for A, C, T, and G */
+        for (Nucleotide base : BASES) {
+            baseQualities.put(base, new ArrayList<>());
+        }
+
+        for (final SamLocusIterator.RecordAndOffset rec : locus.getRecordAndPositions()) {
+            Nucleotide currentBase = Nucleotide.valueOf(rec.getReadBase());
+            ArrayList<BaseQuality> currentBaseQualityList = baseQualities.get(currentBase);
+            if (currentBaseQualityList == null) { /* this can occur if a non-ACTG read is found */
```

Comment:
The code would be more neat if you initialize the map with empty list one per nucleotide so you can avoid the if in the loop. 

Alternatively you could try to use Collectors.groupingBy to create such a map-list.